analyses/sex-prediction-from-RNASeq/sex-prediction-from-RNASeq.Rmd

```diff
@@ -0,0 +1,167 @@
+---
+title: :"Sex Prediction from RNA-Seq"
+author: Bill Amadio
+date: 2019
+---
+Library calls
+```{r}
+
+library(stats)
+library(glmnet)
+library(glmnetUtils)
+
+```
+
+
+RNA-Seq data is in pbta-gene-expression-kallisto.rds.
+
+```{r}
+
+ge <- readRDS("../../data/pbta-gene-expression-kallisto.stranded.rds")
+
+
+```
+
+Metadata is in pbta-histologies.tsv.  reported_gender is our target variable. Kids_First_Biospecimen_ID is the unique identifier for each sample. 
+
+```{r}
+histologies <- read.delim("../../data/pbta-histologies.tsv", header=TRUE, sep="\t", stringsAsFactors = FALSE)
+
+missing_reported_gender_samples <- histologies[which(histologies[, "reported_gender"] == ""), "Kids_First_Biospecimen_ID"]
+
+```
+
+
+Let's try a model with the top 25% highest mads.
+
+```{r}
+tail_percent <- 0.75
+
+# dropping the gene and transcript identifiers and creating a matrix
+
+gene_expression_mat <- as.matrix(ge[ , -c(1,2)])
+rownames(gene_expression_mat) <- ge[ , 1]
+
+# calculate mean absolute variance for each transcript
+
+tx_mads <- apply(gene_expression_mat, 1, function(x) mad(x, high = TRUE))
+df <- t(gene_expression_mat[which(tx_mads >= quantile(tx_mads, tail_percent)), ])
+
+```
+
+Eliminate rows of df corresponding to samples with missing reported_gender.
+
+```{r}
+
+df <- df[!rownames(df) %in% missing_reported_gender_samples, ]
+
+```
+
+Use reported_gender column from histologies as the response.
+
+Extract reported_gender values from reported_gender column of histologies.  Put these value in a two-column dataframe
+  c("Kids_First_Biospecimen_ID", "reported_gender").
+
+```{r}
+
+reported_gender <- histologies[histologies$Kids_First_Biospecimen_ID %in% rownames(df), c("Kids_First_Biospecimen_ID", "reported_gender")]
+
+```
+
+Check sequence of rownames(df) and reported_gender[, 1].  reported_gender_response holds the reported_gender values in the same Kids_First_Biospecimen_ID sequence as rownames(df).
+
+```{r}
+
+match_index <- unlist(sapply(rownames(df), function(x) which(reported_gender[, 1] == x)))
+
+reported_gender_response <- reported_gender[match_index, 2]
+```
+
+Some values in reported_gender_response = "Not Available".  Eliminate from both reported_gender_response and df.
+
+```{r}
+reported_gender_response_hold <- reported_gender_response
+reported_gender_response <- reported_gender_response[reported_gender_response != "Not Available"]
+df <- df[reported_gender_response_hold != "Not Available", ]
+```
+
+
+Build predictive model for reported_gender_response.
+Hold out 30% of the samples to test accuracy after cross-validation tuning of 
+  lambda and alpha regularization parameters.
+
+```{r}
+
+train_percent <- 0.70
+set.seed(36354)
+
+train_set <- sample(1:nrow(df), floor(train_percent*nrow(df)))
+test_set <- setdiff(1:nrow(df), train_set)
+
+ptm <- proc.time()
+sex.cva <- cva.glmnet(df[train_set, ], reported_gender_response[train_set], standardize=TRUE, 
```

Comment:
It looks like the default for the `nfolds` argument of `cva.glmnet` is `10`. Mostly noting it here for the sake of documentation.

Reply:
OK.